Added import test, fixed secondary structure bug.

mathuin opened the pull request:
There were no tests for importing proteins into the database, so I wrote one.  The test successfully detected that the database was being populated with incorrect values for secondary structure.  The secondary structure code has been corrected, and the test now passes.

The import test should be expanded in the future to cover other import-related issues.

mathuin committed:
Added import test, fixed secondary structure bug.

There were no tests for importing proteins into the database, so I wrote one.  The test successfully detected that the database was being populated with incorrect values for secondary structure.  The secondary structure code has been corrected, and the test now passes.

The import test should be expanded in the future to cover other import-related issues.

pop commented on pgd_splicer/tests.py:
```diff
@@ -329,3 +329,49 @@ def test_crosscheck_cullpdb(self):
+
+        # Build pdbs variable.
+        # NB: structure taken from process_args() in ProcessPDBTask.main()
+        pdbs = [
```
We should get rid of this in favor of passing the `line` to `task.work(...)`. This just looks like madness. We're already defining behavior in `ProcessPDBTask.main()` and passing the line to `task.work` should de-duplicate code well.

This would require some back-end changes but I think in the long run it's worth it to improve code instead of covering it with duct-tape and weird variations of off-white paint.

mathuin replied:
Should be a simple matter of moving the conversion between selection-file line and per-file data structure further down the chain.  It might be possible at another time to eliminate the conversion entirely if we can identify a single method that requires all the details, but that's a rabbit hole I want to avoid today.

mathuin committed:
Removed process_args.

The main() had a function process_args() which converted the selection file line into a hash.  This internal function interfered with testing so the functionality was moved further down the call chain to the workhorse() method.  At a future time, it may be appropriate to push the conversion further down, but for now, this is a good fix.

pop commented on pgd_splicer/tests.py:
```diff
@@ -329,3 +329,36 @@ def test_crosscheck_cullpdb(self):
+            line = f.readline()
+
+        # Install that protein into the database with ProcessPDBTask.
+        pdbs = [line]
```
~~pdbs is set to a list containing a single string because...?~~
EDIT: nevermind

ElijahCaine commented:
:+1: Here's hoping this doesn't break something else.
